### PR TITLE
Data_Engine: Create methods for LogicalAnd and LogicalOr requests added

### LINKED ISSUE #1459: Data_Engine: Add Create methods for LogicalAnd and LogicalOr requests
#### Description:
As in the header, following [BHoM #688](https://github.com/BHoM/BHoM/pull/688).

### PR DESCRIPTION
**Pull request by pawelbaran**
### NOTE: Depends on 
[BHoM #688](https://github.com/BHoM/BHoM/pull/688)

   
### Issues addressed by this PR

Closes #1459

### Test files

Not added as the code is self-explanatory.

### CODE REVIEW

**Comment by pawelbaran** [comment]
@FraserGreenroyd thanks, my bad - fixed!

**Comment by pawelbaran** [comment]
Compliance cleanup done, could you please re-review @FraserGreenroyd and merge if approved? Thanks!

File: Data_Engine/Create/LogicalAndRequest.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+namespace BH.Engine.Data
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        public static LogicalAndRequest LogicalAndRequest(IRequest request1, IRequest request2)
```
Comment (FraserGreenroyd) [inline Data_Engine/Create/LogicalAndRequest.cs]: Can we have documentation on this method

File: Data_Engine/Create/LogicalOrRequest.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+namespace BH.Engine.Data
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        public static LogicalOrRequest LogicalOrRequest(IRequest request1, IRequest request2)
```
Comment (FraserGreenroyd) [inline Data_Engine/Create/LogicalOrRequest.cs]: Can we have documentation on this method

Comment (pawelbaran) [reply Data_Engine/Create/LogicalOrRequest.cs]: Fixed

File: Data_Engine/Create/LogicalAndRequest.cs (hunk 1 above)
Comment (pawelbaran) [reply Data_Engine/Create/LogicalAndRequest.cs]: Fixed

File: Data_Engine/Create/LogicalAndRequest.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Creates an IRequest that combines a couple of requests into a logical AND statement.")]
+        [Input("request1", "First request in logical statement")]
+        [Input("request2", "Second request in logical statement")]
+        [Output("LogicalOrRequest")]
```
Comment (FraserGreenroyd) [inline Data_Engine/Create/LogicalAndRequest.cs]: Isn't this meant to be `LogicalAndRequest`?

Comment (pawelbaran) [reply Data_Engine/Create/LogicalAndRequest.cs]: Busted! Sorry for that one, fix pushed :)

**Pull request merged**